src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -388,38 +417,66 @@ public static boolean isPathIdentifier(String name) {
     return name.contains(MODULE_SLASH);
   }
 
+  /** Removes the leading slash from absolute identifiers */
+  public static String removeAbsoluteIdentifierIndicator(String name) {
+    if (isAbsoluteIdentifier(name)) {
+      return name.substring(1);
+    }
+    return name;
+  }
+
+  /** Converts paths which don't start with either a '.' or '/' to absolute paths */
+  public static String toAbsoluteIdentifier(String name) {
+    if (name.length() == 0 || isAbsoluteIdentifier(name) || isRelativeIdentifier(name)) {
+      return name;
```

Comment:
Wait, this can return a relative identifier?
That's just misleading.
I saw your post regarding troublesome terminology, though I cannot find it now.
I propose this.

- "absolute" means "starts with '/'"
- "relative" means "starts with './'"
- "ambiguous" means everything else.

How about we detect ambiguous paths ASAP and either report an error or explicitly force them to be relative or absolute at that point?

I'd rather see some duplicated logic that makes it clear what is happening than a "fix it if it is broken, otherwise leave it alone" method like this one.